### PR TITLE
[Merged by Bors] - Update external deps

### LINKED ISSUE #1706: Update deps
## Description

Before sec review we should go and ensure everything is as up to date as possible. See https://github.com/kbknapp/cargo-outdated

### PR DESCRIPTION
**Pull request by paulhauner**
## Issue Addressed

- Resolves #1706 

## Proposed Changes

Updates dependencies across the workspace. Any crate that was not able to be brought to the latest version is listed in #1712.

## Additional Info

NA

### CODE REVIEW

File: beacon_node/http_api/Cargo.toml [hunk 1]
```diff
@@ -30,5 +30,5 @@ slot_clock = { path = "../../common/slot_clock" }
 [dev-dependencies]
 store = { path = "../store" }
 environment = { path = "../../lighthouse/environment" }
-tree_hash = { path = "../../consensus/tree_hash" }
-discv5 = { version = "0.1.0-alpha.10", features = ["libp2p"] }
+tree_hash = "0.1.0"
```
Comment (realbigsean) [inline beacon_node/http_api/Cargo.toml:33]: Looks like the `tree_hash` in the code base is 0.1.1, not sure what the difference here is though.

File: common/validator_dir/Cargo.toml [hunk 2]
```diff
@@ -13,10 +13,10 @@ insecure_keys = []
 bls = { path = "../../crypto/bls" }
 eth2_keystore = { path = "../../crypto/eth2_keystore" }
 types = { path = "../../consensus/types" }
-rand = "0.7.2"
+rand = "0.7.3"
 deposit_contract = { path = "../deposit_contract" }
-rayon = "1.3.0"
-tree_hash = { path = "../../consensus/tree_hash" }
+rayon = "1.4.1"
+tree_hash = "0.1.0"
```
Comment (realbigsean) [inline common/validator_dir/Cargo.toml:19]: Same as above

File: consensus/fork_choice/Cargo.toml [hunk 3]
```diff
@@ -9,13 +9,13 @@ edition = "2018"
...
-eth2_ssz_derive = { path = "../ssz_derive" }
+eth2_ssz = "0.1.2"
+eth2_ssz_derive = "0.1.0"
 
 [dev-dependencies]
 state_processing = { path = "../../consensus/state_processing" }
 beacon_chain = { path = "../../beacon_node/beacon_chain" }
 store = { path = "../../beacon_node/store" }
-tree_hash = { path = "../../consensus/tree_hash" }
+tree_hash = "0.1.0"
```
Comment (realbigsean) [inline consensus/fork_choice/Cargo.toml:19]: Same as above

File: validator_client/slashing_protection/Cargo.toml [hunk 4]
```diff
@@ -7,15 +7,15 @@ edition = "2018"
 [dependencies]
 tempfile = "3.1.0"
 types = { path = "../../consensus/types" }
-tree_hash = { path = "../../consensus/tree_hash" }
-rusqlite = { version = "0.23.1", features = ["bundled"] }
-r2d2 = "0.8.8"
-r2d2_sqlite = "0.16.0"
+tree_hash = "0.1.0"
```
Comment (realbigsean) [inline validator_client/slashing_protection/Cargo.toml:10]: Same

**Review by realbigsean — COMMENTED** [review]
Just had one comment (in a few places) on the `tree_hash` dependency. Looks like it went from using a relative path (to version 0.1.1) to now using what's on crates.io (0.1.0). Not sure if this was unintentional.

**Comment by AgeManning** [comment]
`tree_hash` is patched in the global workspace. I do recall there being a necessity for the patch, but am not sure why we can't just reference by path everywhere and be consistent?

**Comment by paulhauner** [comment]
> I do recall there being a necessity for the patch, but am not sure why we can't just reference by path everywhere and be consistent?

You can't publish on crates.io with a local path reference.

**Review by AgeManning — APPROVED** [review]

**Review by Thales-de-Milet — COMMENTED** [review]
#3130

**Pull request closed without merging**